index.js

```
@@ -71,22 +76,33 @@ function AtImport(options) {
 
     var hashFiles = {}
 
-    parseStyles(
+    return parseStyles(
       result,
       styles,
       options,
-      insertRules,
       importedFiles,
       ignoredAtRules,
       null,
-      hashFiles
-    )
-    addIgnoredAtRulesOnTop(styles, ignoredAtRules)
+      hashFiles,
+      createProcessor(result, options.plugins)
+    ).then(function() {
+      addIgnoredAtRulesOnTop(styles, ignoredAtRules)
 
-    if (typeof options.onImport === "function") {
-      options.onImport(Object.keys(importedFiles))
+      if (typeof options.onImport === "function") {
+        options.onImport(Object.keys(importedFiles))
+      }
+    })
+  }
+}
+
+function createProcessor(result, plugins) {
+  if (plugins) {
+    if (Array.isArray(plugins)) {
+      return postcss(plugins)
     }
+    result.warn("plugins must be an array")
```

We should probably throw an error instead for this.